ResearchKit/Common/ORKAnswerFormat.m

```diff
@@ -1887,39 +1908,40 @@ - (instancetype)copyWithZone:(NSZone *)zone {
 - (BOOL)isAnswerValid:(id)answer {
     BOOL isValid = NO;
     if ([answer isKindOfClass:[NSString class]]) {
-        return [self isAnswerValidWithString:(NSString *)answer];
+        isValid = [self isAnswerValidWithString:(NSString *)answer];
     }
     return isValid;
 }
 
 - (BOOL)isAnswerValidWithString:(NSString *)text {
-    if (text.length > 0) {
-        return ([self isTextLengthValidWithString:text] && [self isEmailAddressValidWithString:text]);
+    if (text) {
+        BOOL isValid = NO;
+        if (self.regex) {
+            NSString *regExPattern = self.regex;
+            NSRegularExpression *regEx = [[NSRegularExpression alloc] initWithPattern:regExPattern options:NSRegularExpressionCaseInsensitive error:nil];
+            NSUInteger regExMatches = [regEx numberOfMatchesInString:text options:0 range:NSMakeRange(0, [text length])];
+            isValid = (regExMatches != 0);
+        } else if (text.length > 0) {
+	        isValid = [self isTextLengthValidWithString:text];
```

Comment:
Let's re-factor this to be cleaner and more complete. Since we changed `maximumLength` from *readonly* to *readwrite*, we need to run `isTextLengthValidWithString:` on a regex input.
```
if (text.length > 0) {
     isValid = [self isTextLengthValidWithString:text] && [self isTextRegexValidWithString:text];
}
```

And then move the code for regex validation inside the function called `isTextRegexValidWithString:`.
```
- (BOOL)isTextRegexValidWithString:(NSString *)string {
   BOOL isValid = YES;
   if (self.regex) {
      NSString *regExPattern = self.regex;
      ...
      isValid = (regExMatches != 0);
   }
   return isValid;
}